pubsub/drivertest/drivertest.go

```diff
@@ -705,9 +712,24 @@ func testMetadata(t *testing.T, newHarness HarnessMaker) {
 	}
 	defer h.Close()
 
-	weirdMetadata := map[string]string{}
-	for _, k := range escape.WeirdStrings {
-		weirdMetadata[k] = k
+	// Some provider limit the number of metadata per message.
```

Comment:
We should document this non-portability somewhere. Probably on `Message.Metadata`. The current doc comment reads
```
// Metadata has key/value metadata for the message. It will be nil if the
// message has no associated metadata.
```
which only describes the receive side. It should be changed to describe the send side (e.g. "Before sending, set this ....") and to say that some providers have a max.

Reply:
Updated in `pubsub.go`.